glean-core/src/upload/mod.rs

```diff
@@ -219,6 +225,16 @@ impl PingUploadManager {
         self.processed_pending_pings.load(Ordering::SeqCst)
     }
 
+    /// Gets the ping queue and checks in a ping with a certain document_id is already enqueued.
+    fn has_enqueued(
+        queue: &RwLockWriteGuard<'_, VecDeque<PingRequest>>,
```

Comment:
We should be able to take just a slice here, the function shouldn't care about the fact it's behind a lock.
 
```suggestion
        queue: &VecDeque<PingRequest>,
```